### PR TITLE
feat: add view-only role

### PR DESCRIPTION
**Pull request by KoNekoD**

### CODE REVIEW

File: assets/src/components/header.tsx [hunk 1]
```diff
@@ -37,6 +39,23 @@ export const Header = () => {
...
+        onClick={
+          () => {
+            if (!window.confirm('Are you sure you want to logout?')) {
+              return;
+            }
+
+            useLogoutMutation.mutateAsync(undefined, undefined).then(
+              () => {
+                window.location.reload();
+              },
```
Comment (lerothias) [inline assets/src/components/header.tsx:52]: `() =>  window.location.reload()`

File: assets/src/components/header.tsx [hunk 2]
```diff
@@ -37,6 +39,23 @@ export const Header = () => {
...
+            );
+          }
+        }
```
Comment (lerothias) [inline assets/src/components/header.tsx:55]: separate to `handleLogout`

File: assets/src/shared/lib/notify-error.ts [hunk 3]
```diff
@@ -1,4 +1,8 @@
 import { toastManager } from '~/shared/lib/toastManager';
 
-export const notifyError = (err: any) =>
-  toastManager.error(err.message ?? 'Something went wrong: ' + '\n\n' + err.response.data.detail ?? 'No details');
+export const notifyError = (err: any) => {
+  const heading = err.response.data.title ?? err.message ?? 'Something went wrong: '
+  const details = err.response.data.detail ?? 'No details'
```
Comment (lerothias) [inline assets/src/shared/lib/notify-error.ts:5]: maybe empty string?

**Review by lerothias — CHANGES_REQUESTED** [review]
please check prettier format, + use commands for this

**Comment by lerothias** [comment]
fix pipeline (prettier)

File: assets/src/components/header.tsx (hunk 1 above)
Comment (KoNekoD) [reply assets/src/components/header.tsx]: fixed 79427035c9f730f617575521ef539f5ce429e34f

File: assets/src/components/header.tsx (hunk 2 above)
Comment (KoNekoD) [reply assets/src/components/header.tsx]: done 1981c11fabf1b21a92ead789c950bf23ce067a6a

File: assets/src/shared/lib/notify-error.ts (hunk 3 above)
Comment (KoNekoD) [reply assets/src/shared/lib/notify-error.ts:5]: Hmm

Comment (KoNekoD) [reply assets/src/shared/lib/notify-error.ts:5]: Let's try this

Comment (KoNekoD) [reply assets/src/shared/lib/notify-error.ts]: fix b1654cfaf7acb94c1128dbb025d2b7e1c7ca3f29

**Comment by KoNekoD** [comment]
@lerothis please check again 7d4faf63664c8047ee09418e11fbad70efb88457

**Review by lerothias — APPROVED** [review]
good

**Pull request merged**